common/src/main/java/us/myles/ViaVersion/ViaManager.java

```diff
@@ -47,8 +47,23 @@ public void init() {
         // Check for updates
         if (platform.getConf().isCheckForUpdates())
             UpdateUtil.sendUpdateMessage();
-        // Force class load
-        ProtocolRegistry.getSupportedVersions();
+
+        //Register Protocols
+        ProtocolRegistry.registerProtocols();
+
+        //Load Mapping data
+        if(!platform.getConf().isLoadMappingsAsync()){
+            Via.getPlatform().getLogger().info("Loading mapping data ...");
+            ProtocolRegistry.loadMappings();
+        }else{
+            Via.getPlatform().getLogger().info("Load mapping data async ...");
+            Thread thread = new Thread(() -> {
```

Comment:
I think using CompletableFuture.wait and a runnable for each protocol would be more interesting

Reply:
Possiblty, that would then also need proper "api" for ViaBackwards usage

Reply:
Do you think it makes a significant difference then, because it still blocks because it was waiting for the mappings to load

Reply:
You can use a Multi threaded executor for the CompletableFuture